Compilation fails for Plasma 5.18.1

OS Information:
KDE Neon 5.18.1 (Ubuntu 18.04)

Installed packages:
* libkf5config-dev 
* libkf5coreaddons-dev 
* libkf5guiaddons-dev
* extra-cmake-modules

Running the build (with all dependencies installed), results in the following errors:

```make
$ make
Scanning dependencies of target materialdecoration_autogen
[ 11%] Automatic MOC and RCC for target materialdecoration
[ 11%] Built target materialdecoration_autogen
Scanning dependencies of target materialdecoration
[ 22%] Building CXX object src/CMakeFiles/materialdecoration.dir/BoxShadowHelper.cc.o
[ 33%] Building CXX object src/CMakeFiles/materialdecoration.dir/CloseButton.cc.o
/home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.cc: In constructor ‘Material::CloseButton::CloseButton(Material::Decoration*, QObject*)’:
/home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.cc:34:55: error: ‘T* QWeakPointer<T>::data() const [with T = KDecoration2::DecoratedClient]’ is deprecated: Use toStrongRef() instead, and data() on the returned QSharedPointer [-Werror=deprecated-declarations]
     auto *decoratedClient = decoration->client().data();
                                                       ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qsharedpointer.h:48:0,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qpointer.h:43,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QPointer:1,
                 from /usr/include/KDecoration2/kdecoration2/decorationbutton.h:27,
                 from /usr/include/KDecoration2/KDecoration2/DecorationButton:1,
                 from /home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.h:21,
                 from /home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.cc:19:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qsharedpointer_impl.h:565:8: note: declared here
     T *data() const noexcept { return internalData(); }
        ^~~~
/home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.cc: In member function ‘QColor Material::CloseButton::backgroundColor() const’:
/home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.cc:87:53: error: ‘T* QWeakPointer<T>::data() const [with T = KDecoration2::DecoratedClient]’ is deprecated: Use toStrongRef() instead, and data() on the returned QSharedPointer [-Werror=deprecated-declarations]
         auto *decoratedClient = deco->client().data();
                                                     ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qsharedpointer.h:48:0,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qpointer.h:43,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QPointer:1,
                 from /usr/include/KDecoration2/kdecoration2/decorationbutton.h:27,
                 from /usr/include/KDecoration2/KDecoration2/DecorationButton:1,
                 from /home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.h:21,
                 from /home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.cc:19:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qsharedpointer_impl.h:565:8: note: declared here
     T *data() const noexcept { return internalData(); }
        ^~~~
/home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.cc:95:53: error: ‘T* QWeakPointer<T>::data() const [with T = KDecoration2::DecoratedClient]’ is deprecated: Use toStrongRef() instead, and data() on the returned QSharedPointer [-Werror=deprecated-declarations]
         auto *decoratedClient = deco->client().data();
                                                     ^
In file included from /usr/include/x86_64-linux-gnu/qt5/QtCore/qsharedpointer.h:48:0,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/qpointer.h:43,
                 from /usr/include/x86_64-linux-gnu/qt5/QtCore/QPointer:1,
                 from /usr/include/KDecoration2/kdecoration2/decorationbutton.h:27,
                 from /usr/include/KDecoration2/KDecoration2/DecorationButton:1,
                 from /home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.h:21,
                 from /home/jeffrey/Workspace/github.com/material-decoration/src/CloseButton.cc:19:
/usr/include/x86_64-linux-gnu/qt5/QtCore/qsharedpointer_impl.h:565:8: note: declared here
     T *data() const noexcept { return internalData(); }
        ^~~~
cc1plus: all warnings being treated as errors
src/CMakeFiles/materialdecoration.dir/build.make:86: recipe for target 'src/CMakeFiles/materialdecoration.dir/CloseButton.cc.o' failed
make[2]: *** [src/CMakeFiles/materialdecoration.dir/CloseButton.cc.o] Error 1
CMakeFiles/Makefile2:282: recipe for target 'src/CMakeFiles/materialdecoration.dir/all' failed
make[1]: *** [src/CMakeFiles/materialdecoration.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/src/CloseButton.cc
+++ b/src/CloseButton.cc
@@ -31,7 +31,7 @@ namespace Material
 CloseButton::CloseButton(Decoration *decoration, QObject *parent)
     : DecorationButton(KDecoration2::DecorationButtonType::Close, decoration, parent)
 {
-    auto *decoratedClient = decoration->client().data();
+    auto *decoratedClient = decoration->client().toStrongRef().data();
     connect(decoratedClient, &KDecoration2::DecoratedClient::closeableChanged,
             this, &CloseButton::setVisible);
 
@@ -84,7 +84,7 @@ QColor CloseButton::backgroundColor() const
     }
 
     if (isPressed()) {
-        auto *decoratedClient = deco->client().data();
+        auto *decoratedClient = deco->client().toStrongRef().data();
         return decoratedClient->color(
             KDecoration2::ColorGroup::Warning,
             KDecoration2::ColorRole::Foreground
@@ -92,7 +92,7 @@ QColor CloseButton::backgroundColor() const
     }
 
     if (isHovered()) {
-        auto *decoratedClient = deco->client().data();
+        auto *decoratedClient = deco->client().toStrongRef().data();
         return decoratedClient->color(
             KDecoration2::ColorGroup::Warning,
             KDecoration2::ColorRole::Foreground

--- a/src/Decoration.cc
+++ b/src/Decoration.cc
@@ -98,7 +98,7 @@ Decoration::~Decoration()
 
 void Decoration::paint(QPainter *painter, const QRect &repaintRegion)
 {
-    auto *decoratedClient = client().data();
+    auto *decoratedClient = client().toStrongRef().data();
 
     if (!decoratedClient->isShaded()) {
         paintFrameBackground(painter, repaintRegion);
@@ -111,7 +111,7 @@ void Decoration::paint(QPainter *painter, const QRect &repaintRegion)
 
 void Decoration::init()
 {
-    auto *decoratedClient = client().data();
+    auto *decoratedClient = client().toStrongRef().data();
 
     connect(decoratedClient, &KDecoration2::DecoratedClient::widthChanged,
             this, &Decoration::updateTitleBar);
@@ -191,7 +191,7 @@ void Decoration::updateResizeBorders()
 
 void Decoration::updateTitleBar()
 {
-    auto *decoratedClient = client().data();
+    auto *decoratedClient = client().toStrongRef().data();
     setTitleBar(QRect(0, 0, decoratedClient->width(), titleBarHeight()));
 }
 
@@ -293,7 +293,7 @@ void Decoration::paintFrameBackground(QPainter *painter, const QRect &repaintReg
 {
     Q_UNUSED(repaintRegion)
 
-    const auto *decoratedClient = client().data();
+    const auto *decoratedClient = client().toStrongRef().data();
 
     painter->save();
 
@@ -313,7 +313,7 @@ void Decoration::paintFrameBackground(QPainter *painter, const QRect &repaintReg
 
 QColor Decoration::titleBarBackgroundColor() const
 {
-    const auto *decoratedClient = client().data();
+    const auto *decoratedClient = client().toStrongRef().data();
     const auto group = decoratedClient->isActive()
         ? KDecoration2::ColorGroup::Active
         : KDecoration2::ColorGroup::Inactive;
@@ -327,7 +327,7 @@ QColor Decoration::titleBarBackgroundColor() const
 
 QColor Decoration::titleBarForegroundColor() const
 {
-    const auto *decoratedClient = client().data();
+    const auto *decoratedClient = client().toStrongRef().data();
     const auto group = decoratedClient->isActive()
         ? KDecoration2::ColorGroup::Active
         : KDecoration2::ColorGroup::Inactive;
@@ -338,7 +338,7 @@ void Decoration::paintTitleBarBackground(QPainter *painter, const QRect &repaint
 {
     Q_UNUSED(repaintRegion)
 
-    const auto *decoratedClient = client().data();
+    const auto *decoratedClient = client().toStrongRef().data();
 
     painter->save();
     painter->setPen(Qt::NoPen);
@@ -351,7 +351,7 @@ void Decoration::paintCaption(QPainter *painter, const QRect &repaintRegion) con
 {
     Q_UNUSED(repaintRegion)
 
-    const auto *decoratedClient = client().data();
+    const auto *decoratedClient = client().toStrongRef().data();
 
     const int textWidth = settings()->fontMetrics().boundingRect(decoratedClient->caption()).width();
     const QRect textRect((size().width() - textWidth) / 2, 0, textWidth, titleBarHeight());

--- a/src/MaximizeButton.cc
+++ b/src/MaximizeButton.cc
@@ -34,7 +34,7 @@ namespace Material
 MaximizeButton::MaximizeButton(Decoration *decoration, QObject *parent)
     : DecorationButton(KDecoration2::DecorationButtonType::Maximize, decoration, parent)
 {
-    auto *decoratedClient = decoration->client().data();
+    auto *decoratedClient = decoration->client().toStrongRef().data();
     connect(decoratedClient, &KDecoration2::DecoratedClient::maximizeableChanged,
             this, &MaximizeButton::setVisible);
 

--- a/src/MinimizeButton.cc
+++ b/src/MinimizeButton.cc
@@ -34,7 +34,7 @@ namespace Material
 MinimizeButton::MinimizeButton(Decoration *decoration, QObject *parent)
     : DecorationButton(KDecoration2::DecorationButtonType::Minimize, decoration, parent)
 {
-    auto *decoratedClient = decoration->client().data();
+    auto *decoratedClient = decoration->client().toStrongRef().data();
     connect(decoratedClient, &KDecoration2::DecoratedClient::minimizeableChanged,
             this, &MinimizeButton::setVisible);
 

